kmk/internal_state.py

```diff
@@ -12,13 +12,124 @@ class InternalState:
     leader_last_len = 0
     hid_pending = False
     leader_mode_history = []
+    current_key = ""
+    prev_key = ""
+    keylog_update = False
+    prev_key = ""
+    prev_layer_name = "BASE"
+    layer_names = []
+    key_log = []
+    prev_key_string = ""
+    prev_key_log = []
+    # mode badge handling
+    prev_active_layer = []
+    mode_badge = []
+    prev_mode_badge = []
+    prev_logo = []
+
+    alphas = {
```

Comment:
I haven't finished reading the whole PR at the time of writing this comment, so maybe it's answered elsewhere, but on a first read: why this instead of reusing what's in `kmk/keys.py`?